password/password.cabal

```diff
@@ -46,6 +46,11 @@ extra-source-files:
     README.md
     ChangeLog.md
 
+Flag internal
+  description: Export internal module (for testing purposes)
+  default: False
+  manual: True
```

Comment:
I'm generally a fan of just exporting `.Internal` modules with the thinking that users know not to use them, and if they do, then they _really_ need to know what they are doing.

But I've never seen this idea of putting a module behind a cabal flag (although it looks like you were mostly just doing this for testing)!  Thought it was interesting.

I don't really feel strongly either way, but I'd also be fine with just exporting the module and not having the cabal flag.

Reply:
I prefer to not show `Internal` modules, but make sure that users can do everything that they could want to do.
So `Internal` modules only containing stuff that's convenience for implementing the other modules, not actual (potential) functionality.

I think using doctests to test the internal stuff might be a good idea, as @blackheaven suggested. I'll try that out.